File: nodejs/answers/9095.js
const input = require("fs")
  .readFileSync("test.txt")
  .toString()
  .trim()
  .split("\n")
  .map(Number);
const num = input.shift();
const dp = new Array(11).fill(0);

dp[1] = 1;
dp[2] = 2;
dp[3] = 4;

for (let i = 4; i < 11; i++) {
  dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3];
}

input.forEach((i) => {
  console.log(dp[i]);
});

File: python/9095.py
t = int(input())

solve = [0 for i in range(0,12)]
solve[1] = 1
solve[2] = 2
solve[3] = 4

for i in range(12):
    if i<4: continue
    solve[i] = solve[i-1] + solve[i-2] + solve[i-3]

for _ in range(t):
    num = int(input())
    print(solve[num])
